packages/epics/src/membership/hooks/use-member-details.ts

```diff
@@ -0,0 +1,47 @@
+'use client';
+import { useState, useEffect, useCallback } from 'react';
+import { AgreementItem } from '@hypha-platform/graphql/rsc';
+
+export const useMemberDetails = (agreements: AgreementItem[]) => {
+  const [activeFilter, setActiveFilter] = useState<string>('all');
+  const [pages, setPages] = useState(1);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const agreementsPerPage = 4;
+
+  const filteredAgreements = agreements.filter((agreement) =>
+    activeFilter === 'all' ? true : agreement.status === activeFilter
+  );
+
+  const paginatedAgreements = filteredAgreements.slice(
+    0,
+    pages * agreementsPerPage
+  );
+
+  const totalPages = Math.ceil(filteredAgreements.length / agreementsPerPage);
```

Comment:
Please familiarise yourself with the use of the correct react hooks. This will create unnecessary rerenders.

Reply:
Fixed